From 88e75363ff9c00cc2e7768ca23ded79bf8d6bf08 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke@hauke-m.de>
Date: Fri, 7 Feb 2014 19:13:11 +0100
Subject: [PATCH 1/8] backports: add led_trigger_blink{_oneshot}()

When led support is deactivated in the kernel and
CPTCFG_BACKPORT_BUILD_LEDS is set mac80211 references
led_trigger_blink_oneshot() but it is not declared anywhere.

This fixes the following build error:
net/mac80211/led.c: In function 'ieee80211_led_rx':
net/mac80211/led.c:25:2: error: implicit declaration of function 'led_trigger_blink_oneshot' [-Werror=implicit-function-declaration]

Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
---
 backport-include/backport/leds-disabled.h |   13 +++++++++++++
 1 file changed, 13 insertions(+)

--- a/backport-include/backport/leds-disabled.h
+++ b/backport-include/backport/leds-disabled.h
@@ -176,6 +176,19 @@ static inline void led_trigger_blink_one
 					     int invert)
 {
 }
+
+static inline void led_trigger_blink(struct led_trigger *trigger,
+				     unsigned long *delay_on,
+				     unsigned long *delay_off)
+{
+}
+
+static inline void led_trigger_blink_oneshot(struct led_trigger *trigger,
+					     unsigned long *delay_on,
+					     unsigned long *delay_off,
+					     int invert)
+{
+}
 #endif
 
 #endif /* __BACKPORT_LED_DISABLED_SUPPORT */
